gkeys/gkeys/actions.py

```diff
@@ -490,19 +490,19 @@ def speccheck(self, args):
                         uid = ''
                         if gkey.uid:
                             uid = gkey.uid[0]
-                            match = re.findall(r'[\w\.-]+@[\w\.-]+', uid)
-                            uid = ''
-                            if match:
-                                uid = _unicode(match[0])
+                            uid = self.keyhandler._find_email(uid)
+                            if not uid.endswith(self.config.get_key('prefered_address')):
+                                self.logger.debug(_unicode('Email didnt pass the qualifications'))
+                                uid = ''
```

Comment:
this is not done correctly, see my comments at the _find_email().  This method assumes the preferred email is the first uid found or it errors.  Many devs have several uid's and the @gentoo.org one is rarely the first one in there.